### PR TITLE
Add base dockerfile

### PR DESCRIPTION
**Pull request by ChaamC**
Adds the Dockerfile and environment.yml files for the base image, which are trimmed down files taken from pavics-e2e-workflow-tests.
Note that environment.yml has been split into multiple files, in order to reduce RAM usage when building the image, because of a CondaMemoryError when building the image on Docker Hub.

### CODE REVIEW

**Review by dbyrns — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,17 @@ Fixes:
 ------
 - ...
 
+0.1.0 (2020-02-19)
```
Comment (matprov) [inline CHANGELOG.md]: should be 2021 !

Comment (ChaamC) [reply CHANGELOG.md]: Oups hahah >_< Same error in the other PR.

**Review by matprov — APPROVED** [review]
LGTM, only need to fix changelog year.
I haven't deeply looked at the dependencies since it should pretty much be the same thing as pavics-workflow. Once it builds and deps are in sync with the future Confluence page (see https://github.com/crim-ca/pavics-jupyter-images/pull/2) containing the deps list, should be all good.

Thanks for having looked into the CondaMemoryError.

File: Dockerfile [hunk 2]
```diff
@@ -1,3 +1,61 @@
...
+    && useradd --uid 1000 --gid jenkins --create-home jenkins
+
+# Change these folders' permissions for jupyter-conda extension
+RUN chmod -R a+rwx /opt/conda
+
+# create env "birdy"
+# environment is split into multiple .yml files, in order to reduce the amount of RAM usage for each Dockerfile steps
+# and to avoid CondaMemoryError when building the image on Docker Hub
+# use umask 0000 so that the files for the new environment are usable by user 'jenkins' for the jupyter-conda-extension
+RUN umask 0000 && conda env create -f /environment/environment_main.yml \
```
Comment (tlvu) [inline Dockerfile:23]: Oh nice, all in the same layer to reduce disk space waste.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,8 @@
 # pavics-jupyter-base
 Base Jupyter docker image for PAVICS.
+Specialized component images will be built from this image for both CRIM and Ouranos
+organizations.
+
+Repos for the component images :
+* crim-ca/pavics-jupyter-images : https://github.com/crim-ca/pavics-jupyter-images
+* Ouranos/pavics-jupyter-images : to be created
```
Comment (tlvu) [inline README.md]: Our org is https://github.com/Ouranosinc (missing "inc") at the end.

File: environment/environment_main.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# This file contains the main required packages (such as jupyter) and other generic utilities
+name: birdy
+channels:
+  - conda-forge
+  - defaults
+dependencies:
+  # avoid owslib 0.22 with timeout bug
+  # https://github.com/geopython/OWSLib/issues/737
+  - owslib==0.21.0
```
Comment (tlvu) [inline environment/environment_main.yml]: I've unpinned owslib, see https://github.com/Ouranosinc/PAVICS-e2e-workflow-tests/pull/61

**Review by tlvu — APPROVED** [review]
LGTM, very minor changes needed.

Is there a corresponding DockerHub build for this branch, just curious of the size and build speed.

**Comment by ChaamC** [comment]
@tlvu 
There will be a new build with the minor changes (fix Ouranosinc name, and put owslib>=0.23.0), but if I look at the latest one that was built today, it took about 30 minutes ( [Build in 'add-base-dockerfile' (32cbb13b)](https://hub.docker.com/repository/registry-1.docker.io/birdhouse/pavics-jupyter-base/builds/bedb9408-534d-4295-8b7d-da5f74350d57) ) and the image has a compressed size of 1.62 GB ( [DockerHub Image](https://hub.docker.com/layers/138316488/birdhouse/pavics-jupyter-base/add-base-dockerfile/images/sha256-2e7f6adbd8ae472e4b0fb037c81e2f4c5bb3ca468d4b3513179748d1f1143110?context=explore) )

**Comment by tlvu** [comment]
+1

**Pull request merged**